src/promptflow/promptflow/_sdk/_utils.py

```diff
@@ -975,9 +978,37 @@ def is_url(value: Union[PathLike, str]) -> bool:
 def is_remote_uri(obj) -> bool:
     # return True if it's supported remote uri
     if isinstance(obj, str):
-        if obj.startswith("azureml:"):
+        if obj.startswith(REMOTE_URI_PREFIX):
             # azureml: started, azureml:name:version, azureml://xxx
             return True
         elif is_url(obj):
             return True
     return False
+
+
+def parse_remote_flow_pattern(flow: object) -> str:
+    # Check if the input matches the correct pattern
+    flow_name = None
+    error_message = (
+        f"Invalid remote flow pattern, got {flow!r} while expecting "
+        f"a remote workspace flow like '{REMOTE_URI_PREFIX}<flow-name>', or a remote registry flow like "
+        f"'{REMOTE_URI_PREFIX}//registries/<registry-name>/models/<flow-name>/versions/<version>'"
+    )
+    if not isinstance(flow, str) or not flow.startswith(REMOTE_URI_PREFIX):
+        raise UserErrorException(error_message)
+
+    # check for registry flow pattern
+    if flow.startswith(REGISTRY_URI_PREFIX):
+        pattern = r"azureml://registries/.*?/models/(?P<name>.*?)/versions/(?P<version>.*?)$"
+        match = re.match(pattern, flow)
+        if not match or len(match.groups()) != 2:
+            raise UserErrorException(error_message)
+        flow_name, _ = match.groups()
+    # check for workspace flow pattern
+    elif flow.startswith(REMOTE_URI_PREFIX):
+        pattern = r"azureml:(?P<name>.*?)$"
+        match = re.match(pattern, flow)
+        if not match or len(match.groups()) != 1:
+            raise UserErrorException(error_message)
+        flow_name = match.groups()[0]
```

Comment:
if you use group name, please use the name.
or else you don't need to use the group name.